src/main/kotlin/org/rust/ide/lineMarkers/RsImplsLineMarkerProvider.kt

```diff
@@ -42,21 +42,44 @@ class RsImplsLineMarkerProvider : LineMarkerProvider {
                 .setTargets(targets)
                 .setPopupTitle("Go to implementation")
                 .setTooltipText("Has implementations")
+                .setCellRenderer(GoToImplRenderer())
                 .createLineMarkerInfo(el)
 
             result.add(info)
         }
     }
 
     companion object {
-        fun implsQuery(psi: PsiElement): Query<RsImplItem>? {
+        private class GoToImplRenderer: DefaultPsiElementCellRenderer() {
```

Comment:
Can be moved outside of the companion object =D
(just to RsImplsLineMarkerProvider nested class)